daml-foundations/daml-ghc/ide/src/Development/IDE/State/Rules/Daml.hs

```diff
@@ -405,10 +404,10 @@ ofInterestRule = do
         let scenarioFiles = files `Set.union` vrFiles
         gc scenarioFiles
         -- compile and notify any errors
-        let runScenarios file = void $ runExceptT $ do
-                world <- lift $ worldForFile file
-                vrs <- useE RunScenarios file
-                lift $ forM vrs $ \(vr, res) -> do
+        let runScenarios file = void $ actionToMaybe $ do
+                world <- worldForFile file
+                vrs <- use_ RunScenarios file
```

Comment:
Can we not use `use` here, and avoid the `actionToMaybe`?

Reply:
Good catch, changed.